docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -12,56 +12,69 @@ sidebar_position: 0.5
-- **Ensure that the minimum version of Grafana is correct.** Make sure that the `grafanaDependency` in your `plugin.json` points to the earliest version of Grafana that your plugin fully supports.
+- **Verify that your data source or app plugin can be provisioned** - Refer to [Provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- **Include default dashboards in your data source or app plugin** - Refer to [Bundling of dashboards](https://grafana.com/developers/plugin-tools/introduction/plugin-types-usage#bundling-of-dashboards) for more information.
+- **Ensure that the minimum version of Grafana is correct** - Make sure that the `grafanaDependency` in your `plugin.json` points to the earliest version of Grafana that your plugin fully supports.
+- **Add linting and auto-completion** - Reduce bugs in your plugin by adding a code snippet like [this one](https://grafana.com/blog/2021/01/21/6-tips-for-improving-your-grafana-plugin-before-you-publish/#tip-3-add-linting-and-auto-completion-to-your-pluginjson) in VS Code to get linting for your plugin.
+- **Include a well-written README** - Give users a deeper understanding of how to configure and use your plugin, but don’t make it essential reading. You want users to be able to understand your plugin intuitively without referring to the documentation if possible.
+- **Allow incremental learning** - Hide advanced options using switches or categories, and let users learn about advanced features when they’re ready.
+- **Get beta testers** - Enlist users in your target audience to try out your plugin before you submit it. Get feedback to help improve your plugin before it's published.
 
 ## Panel plugins
 
-- **Don't store or use credentials.** Panel plugins don't have a way to securely store credentials. If your plugin needs to use credentials, consider using a data source or app plugin instead and a panel plugin to display the information returned by the data source.
-- **Consider creating custom options.** If the default panel options aren't a good fit for what you're trying to offer users, use [custom options](/create-a-plugin/extend-a-plugin/custom-panel-option-editors).
+- **Don't store or use credentials** - Panel plugins don't have a way to securely store credentials. If your plugin needs to use credentials, consider using a data source or app plugin instead and a panel plugin to display the information returned by the data source.
+
+* **Detect query fields automatically** - Consider automatically detecting fields from a user’s query based on field type rather than relying on hard-coded field names. This increases the chance that users will have something useful right away when they switch between visualizations.
```

Comment:
Can you add an example/link here? I think I understand what this means but not 100% sure.

Reply:
I don't have an example to add as this tip comes almost verbatim from the blog post by Marcus Olsson. If it's ambiguous, let's just delete it for now. Deleting.